### DISCUSSION TITLE
Observations and questions about Structurizr onpremises

### DISCUSSION THREAD

**Discussion by MetalDude101** [structurizr/onpremises#95] [General]
I am evaluating Structurizr onpremises for use in a large company. We have dev teams that are each responsible for a "software system". One special team would create\maintain the "enterprise" DSL for views and interactions across systems. We want each dev team to be responsible for their own workspace DSL. The contents of which, would ideally, be generated via CI/CD. The resulting file would create\update in Structurizr onpremises "data" (file folder) location.

I am new to Structurizr,  but this is what i see:
1) Data folders must be numbered. Why? I created a few workspaces via the UI they start with "1". Next one is "2". I manually created a folder in the Data area called "WorkspaceA". It was not recognized by the software. Must a workspace be created only by the UI?
2) I am not sure how Structurizr can recognize (or load) a workspace DSL that was modified behind the scenes by another process ( CI/CD) to update existing components\relations.
3) Is versioning needed? 
4) Must Structurizr convert a DSL file into  workspace.json?
5) Must Structurizr create and use a versioned workspace.json (goes to my point about how can it be modified by a behind the scenes process)

I appreciate all comments and feedback. Thx

**Comment by simonbrowndotje** [comment]
Some quick answers since I have 5 minutes to spare...

1. Workspaces must be created via the UI or via the API (https://docs.structurizr.com/onpremises/admin-api), and the data directory should not be written to directly.
2. It can't, you should use the CLI to push a workspace DSL file to the on-premises installation via the workspace API (https://docs.structurizr.com/onpremises/workspace-api) during your CI/CD process (https://docs.structurizr.com/cli/push).
3. No; versioning can be turned off via the `structurizr.maxWorkspaceVersions` configuration property (https://docs.structurizr.com/onpremises/configuration).
4. No, but the JSON file is the data structure used by the tooling. I'd recommend using the CLI to convert DSL to JSON, via the `push` command.
5. See 3.

I hope that helps, and I'd recommend joining the Slack group to ask how others are using the on-premises installation.

**Reply by MetalDude101** [reply]
Maybe this information could be put into the FAQ? Reading the help areas on DSL and Onpremises did not provide these answers.
